implementations/rust/ockam/ockam_entity/src/traits.rs

```diff
@@ -1,140 +1,65 @@
-use crate::{
-    Contact, ContactsDb, KeyAttributes, ProfileChangeEvent, ProfileEventAttributes,
-    ProfileIdentifier, ProfileSync, TrustPolicy,
-};
-use async_trait::async_trait;
-use ockam_core::{Address, Result, Route};
-use ockam_node::Context;
+use crate::{Changes, Contact, ProfileChangeEvent, ProfileIdentifier};
+use ockam_core::Result;
 use ockam_vault_core::{PublicKey, Secret};
 
-/// Profile identity.
-pub trait ProfileIdentity {
+pub type Proof = Vec<u8>;
+
+/// Identity
+pub trait Identity: Send + 'static {
     /// Return unique [`Profile`] identifier, which is equal to sha256 of the root public key
-    fn identifier(&self) -> Result<ProfileIdentifier>;
-}
+    fn identifier(&mut self) -> Result<ProfileIdentifier>;
```

Comment:
I think part of the function should take immutable self, including this fn

Reply:
I'll work on this

Reply:
Reduced mutability everywhere possible. The remaining ones are due to trickling up mutable vault requirements. However, I'm thinking it might be an anti-pattern to have the same traits at the Public API and internal APIs. It is hamstringing us in a lot of situations I think..So for example, the `Handler` APIs should always be able to be immutable since they are just all messaging.